### ISSUE TITLE
Redirect anonymous bare url to /share page

### ISSUE THREAD

**Issue by maphew** [TriliumNext/Notes#658]
### Discussed in https://github.com/orgs/TriliumNext/discussions/596

<div type='discussions-op-text'>

<sup>Originally posted by **maphew** November 15, 2024</sup>
How to setup Trilium to redirect bare url to the share page, unless the user is logged in? So an anonymous browser arriving at `www.example.net` is redirected to `www.example.net/share` instead of being greeted with the login page.

Or perhaps better, make the share url `www.example.net/` and the "I'm editing" url `www.example.net/edit` ?

preferably using a docker config (so it can be used on PikaPods)</div>

At the technical level, it's actually easy to redirect bare domain to share:

1. In trilium create or choose a note to be your share landing page and set an _owned attribute_ of `#shareRoot`.

![image](https://github.com/user-attachments/assets/be9b809a-87e0-4bcb-9184-ea76486d18ae)

2. In `./src/services/auth.ts` change line 20 from `res.redirect("login")` to `res.redirect("share")`.

```ts
function checkAuth(req: AppRequest, res: Response, next: NextFunction) {
    if (!sqlInit.isDbInitialized()) {
        res.redirect("setup");
    }
    else if (!req.session.loggedIn && !utils.isElectron() && !noAuthentication) {
        #res.redirect("login");
        res.redirect("share");
    }
    else {
        next();
    }
}
```

3. Restart Trilium server

To login, edit url in address bar and replace path with '/login', ex: `https://my.trilium.net/login`

...

This won't work for PikaPods since they only use the lates release stable release from Docker Hub, and making this change in stable needs more thought, work, and documentation.

**Comment by maphew** [comment]
As written above, this would be a major change in out of the box Trilium behaviour, so not to be approached lightly. With my recent learning success in adding a whitelist of html tags to the Options page I think I see a path to how to make this an optional setting too. We'll see how well that goes!

I don't know if this is a good approach architecturally speaking, and welcome comments from experienced minds.

**Comment by perfectra1n** [comment]
Would this change require a user to have a root Share note already set up?:
```json
{
  "message": "Share root note not found"
}
```

I understand the reason why, but IMO since this is firstly a note-taking platform (and not a microblog), I'm not sure if this is the best idea. I understand that your suggestion is what platforms like Wordpress do where the root URL redirects to the website, and `/wp-admin` is reserved for management/editing. I'm sure @meichthys will have a much more sensible response.

You could do exactly what you're suggesting with access to the webserver and changing its configuration, but since you're on Pikapods I don't believe you have access to that.

**Comment by maphew** [comment]
Yes shareRoot needs to be defined, thanks for the reminder! I've added that note to the helper text in the PR

**Comment by maphew** [comment]
Thanks for the feedback @meichthys. Yes it's off by default.

**Comment by hkhiro** [comment]
Hi - exploring and experimenting Trillium just this week and have it self hosted on my NAS. Using Cloudflare Tunnels managed to publish a site. 

While this is a good feature to have was thinking in the IDEAL would want to be able to publish a static site to GitHub repository and use GitHub pages. Is that possible already today?

**Comment by maphew** [comment]
@hhkiro, go to the top node of the branch you wish to use >> 3-dot menu at top right >> Export  note >>  select html in zip >> copy those files to webserver and unzip.

For automation you'd need to write a script using something like [Trilium-py](https://github.com/Nriver/trilium-py) to combine the export and uploading to Github pages (or any other web server).

**Comment by maphew** [comment]
this was implemented in #1207 and is part of trilium v0.92.1-beta and newer.
